### DISCUSSION TITLE
Function order when looping throug global state

### DISCUSSION THREAD

**Discussion by napsy** [yuin/gopher-lua#374] [Q&A]
Hello,

can I iterate through LValue values in the ``_G`` global state in order that they're written in the script. Concretely, I would like to execute functions in the up-to-down order. Is that possible?

**Comment by yuin** [comment]
This is not a GopherLua specific question, so you can do it with pure Lua codes.

```lua
local keys = {}
setmetatable(_G, {
  __newindex = function(t, i, v)
    table.insert(keys, i)
    rawset(t, i, v)
  end
})

function f1()
  print("f1")
end

function f2()
  print("f2")
end

for i, key in ipairs(keys) do
   print(tostring(i) .. ":" .. key)
end
```
Refer to [Lua 5.1 reference manual](http://www.lua.org/manual/5.1/) for details.

**Comment by napsy** [comment]
Cool, can I make the same think using the gopher-lua API? I currently loop through all the objects in ``_G`` using ForEach and checking if the LValue is a function. But the order of the objects is random.

**Comment by napsy** [comment]
I pasted your code into a ``L.DoString()`` after I load my script with ``L.DoFile()`` but the ``keys`` table only fills with two global variables I add and the functions in the script are missing.
